runtime/lua/vim/semver.lua

```diff
@@ -0,0 +1,248 @@
+local M = {}
+
+---@private
+local function cmp_prerelease(v1_parsed, v2_parsed)
+  if v1_parsed.prerelease and not v2_parsed.prerelease then
+    return -1
+  end
+  if not v1_parsed.prerelease and v2_parsed.prerelease then
+    return 1
+  end
+  if not v1_parsed.prerelease and not v2_parsed.prerelease then
+    return 0
+  end
+
+  local v1_identifiers = vim.split(v1_parsed.prerelease, '%.')
+  local v2_identifiers = vim.split(v2_parsed.prerelease, '%.')
+  local i = 1
+  local max = vim.fn.max({ vim.tbl_count(v1_identifiers), vim.tbl_count(v2_identifiers) })
```

Comment:
Better to use plain Lua `:h math.max()`

Reply:
Awesome, thanks. Updated PR with changes.

Reply:
As an aside, what is the reasoning behind using plain lua `math.max()` over `vim.fn.max`?

Reply:
Why wouldn't you use a native function in your programming language rather than a Rube Goldberg machine that goes through three layers of indirection calling into a text editor to compare two numbers? ;)

Reply:
(in other words, performance and robustness)